tests/upgrades/test_Proxy.py

```diff
@@ -1,109 +1,184 @@
+        constructor_calldata=[implementation_decl.class_hash]
     )
     return (
         starknet.state,
-        account,
-        implementation,
+        account1,
+        account2,
         proxy
     )
 
 
 @pytest.fixture
-def proxy_factory(contract_defs, proxy_init):
-    account_def, implementation_def, proxy_def = contract_defs
-    state, account, implementation, proxy = proxy_init
+def proxy_factory(contract_classes, proxy_init):
+    account_cls, _, proxy_cls = contract_classes
+    state, account1, account2, proxy = proxy_init
     _state = state.copy()
-    account = cached_contract(_state, account_def, account)
-    implementation = cached_contract(
-        _state,
-        implementation_def,
-        implementation
+    admin = cached_contract(_state, account_cls, account1)
+    other = cached_contract(_state, account_cls, account2)
+    proxy = cached_contract(_state, proxy_cls, proxy)
+
+    return admin, other, proxy
+
+
+@pytest.fixture
+async def after_initialized(proxy_factory):
+    admin, other, proxy = proxy_factory 
+
+    # initialize proxy
+    await signer.send_transaction(
+        admin, proxy.contract_address, 'initializer', [admin.contract_address]
     )
-    proxy = cached_contract(_state, proxy_def, proxy)
 
-    return account, implementation, proxy
+    return admin, other, proxy
 
+#
+# initializer
+#
 
 @pytest.mark.asyncio
-async def test_constructor_sets_correct_implementation(proxy_factory):
-    account, implementation, proxy = proxy_factory
+async def test_initializer(proxy_factory):
+    admin, _, proxy = proxy_factory 
+
+    await signer.send_transaction(
+        admin, proxy.contract_address, 'initializer', [admin.contract_address]
+    )
 
+    # check admin is set
     execution_info = await signer.send_transaction(
-        account, proxy.contract_address, 'get_implementation', []
+        admin, proxy.contract_address, 'getAdmin', []
     )
-    assert execution_info.result.response == [implementation.contract_address]
+    assert execution_info.result.response == [admin.contract_address]
 
 
 @pytest.mark.asyncio
-async def test_initializer(proxy_factory):
-    account, _, proxy = proxy_factory
+async def test_initializer_after_initialized(after_initialized):
+    admin, _, proxy = after_initialized 
 
-    await signer.send_transaction(
-        account, proxy.contract_address, 'initializer', [
-            account.contract_address]
+    await assert_revert(signer.send_transaction(
+        admin, proxy.contract_address, 'initializer', [admin.contract_address]),
+        reverted_with="Proxy: contract already initialized"
+    )
+
+#
+# set_admin
+#
+
+@pytest.mark.asyncio
+async def test_set_admin(after_initialized):
+    admin, _, proxy = after_initialized 
+
+    # check initial admin
+    execution_info = await signer.send_transaction(
+        admin, proxy.contract_address, 'getAdmin', []
+    )
+    assert execution_info.result.response == [admin.contract_address]
```

Comment:
is this necessary? it looks like we're testing the fixture instead of the library behavior

Reply:
I suppose it's not necessary, more of a sanity check. We can remove it